source/onboard/migrate-from-slack.rst

```diff
@@ -0,0 +1,194 @@
+
+4. Convert Slack import to Mattermost's bulk export format
+----------------------------------------------------------
+
+Now that you have a Slack export file with emails and attachments, let's convert this information into Mattermost's bulk import format using the import preparation tool ``mmetl``. Download the latest release of ``mmetl`` for your `OS and architecture <https://github.com/mattermost/mmetl/releases/>`__. Run ``mmetl help`` to learn more about using the tool.
+
+Next, run the following command to create a Mattermost bulk import file. Replace ``<TEAM NAME>`` with the name of your team in Mattermost:
+
+.. code:: bash
+
+    ./mmetl transform slack --team <TEAM NAME> --file export-with-emails-and-attachments.zip --output mattermost_import.jsonl
+
+The tool outputs a [.jsonl](https://jsonlines.org/examples) file containing all of your users, channels, and posts. It also creates a ``data`` folder that contains all of your attachments.
+
+.. important::
+
+  It doesn't matter what you name the ``.jsonl`` file. You can name it what you want with the `--output` flag as shown above. It just needs to be a ``.jsonl`` file.
+
+The file ``mattermost-bulk-import.zip`` is now ready to import into Mattermost.
+
+5. Import data into Mattermost
+------------------------------
+
+We have two options to run the import process:
+
+1. SSH into the Mattermost server's host, upload the export file to this server's file system somehow, and use the ``mattermost`` command to process the export file.
+2. Uploading the export through Mattermost's API, via command line ``mmctl`` from the server or from another computer. This option is required for Mattermost Cloud deployments.
+
+Note that for the second option, the server will save the import in its file store before running the import (e.g. AWS S3 if you are using that are your file store), so there will be time spent uploading/downloading the file in this case. Since this is really just a one-time thing to import this file, we recommend the first option of running it on the server itself if you have a large export zip file.
+
+Note that the migration is idempotent, meaning that you can run multiple imports that contain the same posts, and there will not be duplicated created posts in Mattermost. Each post is imported with the correct user/author and ``created_at`` value from your Slack instance. Threads are kept in tact with the import.
+
+Option 1: Use the ``mattermost`` command to run the export
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First get the ``.jsonl`` file and ``data`` folder onto your server using something like ``scp``. Then you'll use the ``mattermost`` binary to process the bulk import data:
+
+.. code:: bash
+    mattermost import bulk ./mattermost_import.jsonl --import-path ./data --apply
+
+.. important::
+    If doing this locally in a development environment, you'll want to set the location of your server's data folder first like so:
```

Comment:
I think the "local development environment" scenario might be better suited to an FAQ or "Troubleshooting" section at the bottom.  I would think most readers will be admins looking to do this with official Mattermost releases.

@mickmister thoughts?

Reply:
@sadohert Sounds good to me. Also, maybe all internal testing should be done using S3/Minio and not the local file system setup. If you agree I'll add a piece about recommending external storage over local. There really shouldn't be any production deployments using the local filesystem for data storage anyway. (I've heard the horror stories of data loss)

Reply:
@mickmister - I'll hold off approving this PR so that you can apply this update.